CMakeLists.txt

```diff
@@ -15,25 +15,69 @@ add_subdirectory(source)
 add_subdirectory(3rdparty/software)
 add_subdirectory(3rdparty/plutovg)
 
-if(BUILD_SHARED_LIBS)
-    target_compile_definitions(lunasvg PUBLIC LUNASVG_SHARED)
-    target_compile_definitions(lunasvg PRIVATE LUNASVG_EXPORT)
+set_target_properties(lunasvg
+    PROPERTIES
+    ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib"
+    LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib"
+    RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/bin"
+    CXX_VISIBILITY_PRESET hidden
+    VISIBILITY_INLINES_HIDDEN 1
+)
+
+add_library(lunasvg::lunasvg ALIAS lunasvg)
+
+if(NOT BUILD_SHARED_LIBS)
+    set_target_properties(lunasvg
+        PROPERTIES
+        COMPILE_FLAGS -DLUNARSVG_STATIC_DEFINE
+    )
 endif()
 
 if(LUNASVG_BUILD_EXAMPLES)
     add_subdirectory(example)
 endif()
 
-set(LUNASVG_LIBDIR ${CMAKE_INSTALL_PREFIX}/lib)
-set(LUNASVG_INCDIR ${CMAKE_INSTALL_PREFIX}/include)
+#
+# Installation
+#
+include(GNUInstallDirs)
+install(
+    FILES
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/lunasvg.h
+    ${CMAKE_CURRENT_BINARY_DIR}/include/lunasvg_export.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/lunasvg
+)
+install(
+    TARGETS lunasvg
+    EXPORT lunasvg-targets
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
```

Comment:
IIRC specifying those dirs is redundant/not needed with cmake >=3.14

Reply:
Will keep them then, since we're going with 3.13 as the baseline.